improver/regrid/grid.py

```diff
@@ -382,8 +399,83 @@ def create_regrid_cube(cube_array: ndarray, cube_in: Cube, cube_out: Cube) -> Cu
     cube_v.add_dim_coord(cube_out.coord(cord_2), ndim + 1)
 
     # add all aus_coords from cube_in
+
     for coord in cube_in.aux_coords:
-        dims = np.array(cube_in.coord_dims(coord)) + 1
-        cube_v.add_aux_coord(coord.copy(), dims)
+        cube_v.add_aux_coord(coord.copy(), cube_in.coord_dims(coord))
 
     return cube_v
+
+
+def group_target_points_with_source_domain(
+    cube_in: Cube, out_latlons: ndarray
+) -> Tuple[ndarray, ndarray]:
+    """
+    Group cube_out's grid points into outside or inside cube_in's domain.
+
+    Args:
+        cube_in:
+            Source cube .
+        out_latlons:
+            Target points's latitude-longitudes.
+    Returns:
+        - Index array of target points outside input domain.
+        - Index array of target points inside input domain.
+    """
+
+    # get latitude and longitude coordinates of cube_in
+    lat_coord = cube_in.coord(axis="y").points
+    lon_coord = cube_in.coord(axis="x").points
+
+    in_lat_max, in_lat_min = np.max(lat_coord), np.min(lat_coord)
+    in_lon_max, in_lon_min = np.max(lon_coord), np.min(lon_coord)
+
+    lat = out_latlons[:, 0]
+    lon = out_latlons[:, 1]
+
+    # check target point coordinates inside/outside input source domain
+    in_domain_lat = np.logical_and(lat >= in_lat_min, lat <= in_lat_max)
+    in_domain_lon = np.logical_and(lon >= in_lon_min, lon <= in_lon_max)
+    in_domain = np.logical_and(in_domain_lat, in_domain_lon)
+
+    outside_input_domain_index = np.where(np.logical_not(in_domain))[0]
+    inside_input_domain_index = np.where(in_domain)[0]
+
+    return outside_input_domain_index, inside_input_domain_index
+
+
+def mask_target_points_outside_source_domain(
+    total_out_point_num: int,
+    outside_input_domain_index: ndarray,
+    inside_input_domain_index: ndarray,
+    regrid_result: Union[ndarray, MaskedArray],
+) -> Union[ndarray, MaskedArray]:
+    """
+    Mask target points outside cube_in's domain.
+
+    Args:
+        total_out_point_num:
+            Total number of target points
+        outside_input_domain_index:
+            Index array of target points outside input domain.
+        inside_input_domain_index:
+            Index array of target points inside input domain.
+        regrid_result:
+            Array of regridded result in (lat*lon,....) or (projy*projx,...).
+    Returns:
```

Comment:
Can you add a blank line before `Returns:`.

Reply:
done